pkg/endpoint/log.go

```diff
@@ -25,10 +25,15 @@ var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "endpoint")
 
 // logger returns a logrus object with EndpointID, ContainerID and the Endpoint
 // revision fields.
-// Note: You must hold Endpoint.Mutex
+// Note: You must hold Endpoint.Mutex for reading
 func (e *Endpoint) getLogger() *logrus.Entry {
 	e.updateLogger()
-	return e.logger
+
+	e.loggerMutex.RLock()
+	logger := e.logger
+	e.loggerMutex.RUnlock()
+
+	return logger
 }
 
 // updateLogger creates a logger instance specific to this endpoint. It will
```

Comment:
Could you also update the `// Note: You must hold Endpoint.Mutex` line below, or remove it, so it is consistent with the on on getLogger